In [5]:
from echo.multi_threading_agent_pydantic_second import (find_teams, create_value_prop)
from echo.multi_threading_agent_pydantic_second import get_company_from_url, get_people_data, enrich_people_data, get_influence_score, get_graphs_for_initiatives


import openai
import os
api_key = os.getenv("OPENAI_API_KEY")
client = openai.OpenAI(api_key=api_key)
#client = openai.OpenAI(api_key="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")

buyer = "rippling"
seller = "whatfix"

buyer_initiatives  = """
Buyer is rippling.com. Rippling is a San Francisco-based HR technology company founded in 2016 by Parker Conrad and Prasanna Sankar. It offers a unified platform that integrates HR, IT, and finance functions, enabling businesses to manage employee onboarding, payroll, benefits, device management, and compliance from a single system. The company targets firms with fewer than 2,000 employees and competes with established players like Workday and ADP, as well as startups like Justworks and Deel. As of April 2024,
Rippling achieved a valuation of $13.5 billion after a $200 million funding round led by Coatue, with participation from investors such as Founders Fund, Greenoaks, and Dragoneer .​
Top 3 Goals for 2025:
1. International Expansion: Rippling is focusing on growing its global presence, particularly in India. The company plans to hire over 100 engineers in Bengaluru and expand its workforce across various functions, including product management, design, sales, finance, and customer support .​
2. Investment in Research and Development (R&D): With over $1 billion in cash reserves, Rippling aims to invest heavily in R&D to develop new products and enhance its existing offerings. This includes building tools that support clients and integrating advanced technologies like artificial intelligence to improve workforce management solutions .​
3. Enhancing IT Security and Automation: Recognizing the growing importance of cybersecurity, Rippling is prioritizing IT security by addressing challenges such as data privacy, compliance, and automation of IT processes. The company is working on implementing robust controls, improving identity and device management solutions, and reducing manual, repetitive tasks to minimize human error and enhance overall security .​
These strategic goals align with Rippling's mission to provide a comprehensive, integrated platform that simplifies workforce management for businesses worldwide.​
"""

seller_info = """
​Whatfix is a global leader in digital adoption platforms (DAP),
co-headquartered in San Jose, California, and Bengaluru, India. Founded in 2014 by Khadim Batti and Vara Kumar,
the company offers solutions that enhance user engagement and productivity across web, desktop, and mobile applications.
Serving over 700 enterprises, including more than 80 Fortune 500 companies like Shell, Microsoft, and Schneider Electric,
Whatfix has established a strong presence in the digital transformation space .​


🚧 Key Challenges Addressed by Whatfix
Whatfix's platform is designed to tackle several common challenges faced by organizations:​
Complex Software Onboarding: Traditional onboarding processes can be time-consuming and ineffective. Whatfix provides in-app guidance tools—such as Flows, Task Lists, and Smart Tips—that help users navigate complex applications, reducing the learning curve and enhancing user adoption .​
High Training and Support Costs: Organizations often spend substantial resources on training and support. By offering self-help modules and contextual assistance within applications, Whatfix reduces the need for extensive training sessions and support interventions, leading to significant cost savings .​
Whatfix
Low Feature Adoption: New features in applications may go unnoticed or underutilized. Whatfix addresses this by delivering targeted, in-app prompts and walkthroughs that highlight new functionalities, ensuring users are aware of and can effectively use all features .​
Process Non-Compliance: Ensuring that employees follow standardized processes is crucial for operational efficiency. Whatfix's real-time guidance and analytics help monitor user interactions, ensuring compliance with established workflows and identifying areas for improvement .​
Celent
Inefficient Feedback Mechanisms: Gathering user feedback is essential for continuous improvement. Whatfix enables organizations to collect real-time feedback through in-app surveys and quizzes, providing insights into user experiences and training effectiveness .​
Whatfix

By addressing these challenges, Whatfix empowers organizations to enhance user engagement, streamline training processes, and maximize the return on their technology investments.

"""


In [6]:
seller = 'https://whatfix.com/'
buyer = 'https://rippling.com/'
from echo.indexing import create_tables
from echo.step_templates.utilities.seller_web_search import add_landing_page_data_to_seller_index, add_seller_web_search_data
from echo.utils import json_to_markdown
create_tables(seller)
from echo.step_templates.utilities.buyer_web_search import add_buyer_web_search_data





Created tables for https://whatfix.com/


In [7]:
import json
def return_dummy_content_json():
    f = open("./echo/data/content.json",'r').read()
    return json.dumps(json.loads(f))
print(return_dummy_content_json())
print(type(return_dummy_content_json()))    


{"type": "doc", "content": [{"type": "heading", "attrs": {"textAlign": null, "level": 1}, "content": [{"type": "text", "text": "Getting started"}]}, {"type": "paragraph", "attrs": {"textAlign": null}, "content": [{"type": "text", "text": "Welcome to the "}, {"type": "text", "marks": [{"type": "italic"}, {"type": "highlight", "attrs": {"color": "var(--tt-color-highlight-yellow)"}}], "text": "Simple Editor"}, {"type": "text", "text": " template! This template integrates "}, {"type": "text", "marks": [{"type": "bold"}], "text": "open source"}, {"type": "text", "text": " UI components and Tiptap extensions licensed under "}, {"type": "text", "marks": [{"type": "bold"}], "text": "MIT"}, {"type": "text", "text": "."}]}, {"type": "paragraph", "attrs": {"textAlign": null}, "content": [{"type": "text", "text": "Integrate it by following the "}, {"type": "text", "marks": [{"type": "link", "attrs": {"href": "https://tiptap.dev/docs/ui-components/templates/simple-editor", "target": "_blank", "rel"

In [ ]:
# flow

# seller research

add_landing_page_data_to_seller_index(seller)
add_seller_web_search_data(seller)


# buyer research
buyer_data = add_buyer_web_search_data(buyer=buyer, seller=seller)
from echo.multi_threading_agent_pydantic import account_plan_extract_data
final_data = account_plan_extract_data(buyer_data["full_text"], buyer, seller)
print(final_data.model_dump())


# process discovery call

from echo.step_templates.discovery import BuyerDataExtracted

# fetch discovery transcript from constants or link 
from echo.step_templates.sample_data import discovery_transcript

# fetch discovery parsing prompt
from echo.step_templates.prompts import  DISCOVERY_ANALYSIS_PROMPT


# run through discovery parsing using above pydantic
# create index
# how do we updates?


# get stakeholders, stage
# select top 4 sections or get template based on inputs
# query respective queries and subquery to get the doc

# get the data from the doc



In [14]:
# dummy func 1
from echo.constants import DEMO
from echo.step_templates.prompts import DEMO_BUSINESS_CASE_CONTENT_CHAMPION
def get_content_doc(buyer, seller, doc_type="Business_Case"):
    api_key = os.getenv("OPENAI_API_KEY")
    client = openai.OpenAI(api_key=api_key)




    business_case_user_prompt = f"""    
    the buyer is {buyer} and the seller is {seller}.
    """
    

    response_team_guess_reasoning = client.responses.create(
        model="o4-mini",
        reasoning={"effort": "medium"},
        input=[
            {"role": "system", "content": DEMO_BUSINESS_CASE_CONTENT_CHAMPION},
            {"role": "user", "content": business_case_user_prompt},
        ],
    )
    print(response_team_guess_reasoning)
    return response_team_guess_reasoning
    
#print(DEMO_BUSINESS_CASE_CONTENT_CHAMPION)
a = get_content_doc(buyer, seller, doc_type="Business_Case")

Response(id='resp_6849a08c24448198aac8708af8c6dd560c10e8005886ad32', created_at=1749655692.0, error=None, incomplete_details=None, instructions=None, metadata={}, model='o4-mini-2025-04-16', object='response', output=[ResponseReasoningItem(id='rs_6849a08cfdb88198ac6b6d93c1fa84290c10e8005886ad32', summary=[], type='reasoning', encrypted_content=None, status=None), ResponseOutputMessage(id='msg_6849a0b8bbbc81988e8b1081c91a817b0c10e8005886ad32', content=[ResponseOutputText(annotations=[], text='# High Priority – avoid compliance violations and payroll errors by year-end audit\n\n*Developed by*: **Jordan Lee**, with **IT Operations, HR, Finance**\n\n---\n\n**Headline**\n\nBecause “unifies HR, IT, and Finance operations on a single, modern platform” is reshaping “global workforce management”, we should “simplify workforce management” by implementing Whatfix in Q3 2025. Doing so avoids compliance violations and payroll errors and unlocks “single source of truth” and “automation”.\n\n---\n\n#

In [16]:
print(a.output_text)

# High Priority – avoid compliance violations and payroll errors by year-end audit

*Developed by*: **Jordan Lee**, with **IT Operations, HR, Finance**

---

**Headline**

Because “unifies HR, IT, and Finance operations on a single, modern platform” is reshaping “global workforce management”, we should “simplify workforce management” by implementing Whatfix in Q3 2025. Doing so avoids compliance violations and payroll errors and unlocks “single source of truth” and “automation”.

---

### The Problem Statement

Every month, global HR and finance teams experience manual steps cause delays and errors, costing $250K in rework and compliance fines (*source – internal audit*). If unaddressed by year-end audit, error rates will increase by 15%.

---

### Recommended Approach

- Pain ➜ Capability ➜ Proof  
  1. manual steps cause delays and errors ➜ contextual in-app guidance ➜ [Case Study: Acme Corp](https://whatfix.com/case-study)  
  2. complex software workflows ➜ interactive walkthroughs

# High Priority – avoid compliance violations and payroll errors by year-end audit

*Developed by*: **Jordan Lee**, with **IT Operations, HR, Finance**

---

**Headline**

Because “unifies HR, IT, and Finance operations on a single, modern platform” is reshaping “global workforce management”, we should “simplify workforce management” by implementing Whatfix in Q3 2025. Doing so avoids compliance violations and payroll errors and unlocks “single source of truth” and “automation”.

---

### The Problem Statement

Every month, global HR and finance teams experience manual steps cause delays and errors, costing $250K in rework and compliance fines (*source – internal audit*). If unaddressed by year-end audit, error rates will increase by 15%.

---

### Recommended Approach

- Pain ➜ Capability ➜ Proof  
  1. manual steps cause delays and errors ➜ contextual in-app guidance ➜ [Case Study: Acme Corp](https://whatfix.com/case-study)  
  2. complex software workflows ➜ interactive walkthroughs ➜ [Customer Testimonial](https://whatfix.com/testimonial)  
  3. Sandbox rollout & multilingual support ensure adoption across global HR and finance teams  
• Success factors: Cross-functional team of 3: IT lead, HR lead, Finance analyst; SSO integration; store-level champions

---

### Target Outcomes

| Metric                | Current | Target by year-end audit | Δ     |
|-----------------------|---------|--------------------------|-------|
| Time to proficiency   | 5 days  | 2 days                   | –3 d  |
| Error rate            | 5 %     | 1 %                      | –4 %  |
| Employee satisfaction | 70 %    | 85 %                      | +15 % |

*(Δ = Target − Current when both are numeric.)*

---

### Required Investment

- **Budget**: USD 120 K annual subscription – per user per month $25; payback in < 6 mo.  
- **People & Time**: Cross-functional team of 3: IT lead, HR lead, Finance analyst. Kickoff Q3 2025.  
- **Next Step**: approve charter this week to meet year-end audit.

---

### Evidence & Compliance

*SOC-2 Certificate*  
*ISO 27001 Compliance*
